src/ir/module-utils.cpp

```diff
@@ -339,142 +339,143 @@ struct Counts {
     assert(sig.params.size() == 0);
     if (sig.results.isTuple()) {
       // We have to use a function type.
-      controlFlowSignatures[sig]++;
+      ++controlFlowSignatures[sig];
     } else if (sig.results != Type::none) {
       // The result type can be emitted directly instead of using a function
       // type.
-      note(sig.results[0]);
+      note(sig.results);
```

Comment:
Is this a bugfix?

Reply:
No, this is NFC. We allow indexing into all types for uniformity, but the only valid index for a non-tuple is 0 and the type at index 0 is the original type itself. The indexing was useless extra work here.

Reply:
Sorry, I don't follow. `sig.results` might be a tuple if the signature is a multivalue return, like `(i32 f64)`. Before we just read `i32` in that case?

Reply:
We handle the tuple case separately above. At this point the type is known to not be a tuple.

Reply:
Ah, thanks, that's what I was missing.